tsconfig.json

```diff
@@ -1,6 +1,7 @@
 {
   "compilerOptions": {
     "outDir": "./dist/",
+    "strict": true,
     "sourceMap": true,
     "noImplicitAny": true,
     "module": "commonjs",
```

Comment:
>I've seen that CLUE is using `"module": "ESNext",` but this conflicts with `"resolveJsonModule": true`. We'd need to use webpack JSON loader I guess. However, what's the difference in practice (I've seen in docs different module definition, but wonder what that changes for development or runtime)?

Modern bundling enhancements, e.g. code-splitting, generally require ES modules (`ES6`/`ES2015` or newer). I can't remember exactly why I chose to push it to `ESNext` in CLUE, but it probably had to do with the expectation that we would want to use dynamic imports (as we are now doing on the CLUE 2.2 branch) which was introduced in `ES2020`. According to caniuse, dynamic imports have been [supported by the major browsers](https://caniuse.com/es6-module-dynamic-import) for 3+ years now so it doesn't seem like an unreasonable requirement on the face of it.

I'm not aware of any conflict between `"module": "ESNext"` and `"resolveJsonModule": true`; CLUE is using both of those settings.

Reply:
I also just noticed that this is a "recommended" setting for React applications: https://github.com/tsconfig/bases/blob/main/bases/create-react-app.json.

Reply:
Which setting do you mean? It actually uses `"module": "node"`. It also uses `"lib": ["dom", "dom.iterable", "esnext"],` without listing `es2016`.

> I'm not aware of any conflict between "module": "ESNext" and "resolveJsonModule": true; CLUE is using both of those settings.

I think I saw that somewhere in the docs. But also, if I try to build the project, I can see this error:

> TS5070: Option '--resolveJsonModule' cannot be specified without 'node' module resolution strategy.

Reply:
OK, I see I've been confusing "module" and "moduleResolution" options.

"module": "ESNext" works with "resolveJsonModule": true when "moduleResolution" is set to "node".